tests/jax/components/executing/action_selection.py

```diff
@@ -0,0 +1,237 @@
+    """
+    ff_executor_select_action = FeedforwardExecutorSelectAction(
+        config=dummy_empty_config
+    )
+    assert (
+        ff_executor_select_action.config.__dict__
+        == ExecutorSelectActionConfig().__dict__
+    )
+
+
+# Test on_execution_select_actions
+def test_on_execution_select_actions_with_empty_observations(
+    mock_empty_executor: Executor, dummy_config: type
+) -> None:
+    """Test on_execution_select_actions with empty observations
+
+    Args:
+        mock_empty_executor: executor with no observations and no agents
+    """
+    ff_executor_select_action = FeedforwardExecutorSelectAction(dummy_config)
+    ff_executor_select_action.on_execution_select_actions(executor=mock_empty_executor)
+
+    assert mock_empty_executor.store.actions_info == {}
+    assert mock_empty_executor.store.policies_info == {}
+
+
+def test_on_execution_select_actions(
+    mock_executor: Executor, dummy_config: type
+) -> None:
+    """Test on_execution_select_actions.
+
+    Args:
+        dummy_config: config
+        mock_executor: Executor
+    """
+
+    ff_executor_select_action = FeedforwardExecutorSelectAction(dummy_config)
+    ff_executor_select_action.on_execution_select_actions(executor=mock_executor)
+
+    assert (
+        mock_executor.store.actions_info["agent_0"]
+        == "action_info_agent_0_observation_0"
+    )
+    assert (
+        mock_executor.store.actions_info["agent_1"]
+        == "action_info_agent_1_observation_1"
+    )
+    assert (
+        mock_executor.store.actions_info["agent_2"]
+        == "action_info_agent_2_observation_2"
+    )
+
+    assert (
+        mock_executor.store.policies_info["agent_0"]
+        == "policy_info_agent_0_observation_0"
+    )
+    assert (
+        mock_executor.store.policies_info["agent_1"]
+        == "policy_info_agent_1_observation_1"
+    )
+    assert (
+        mock_executor.store.policies_info["agent_2"]
+        == "policy_info_agent_2_observation_2"
+    )
+
+
+def test_on_execution_select_actions_param(dummy_config: type) -> None:
+    """test errors of parameter enter in on_execution_select_actions
+
+    Args:
+        dummy_config: config
+    """
+    ff_executor_select_action = FeedforwardExecutorSelectAction(dummy_config)
+    with pytest.raises(Exception):
+        ff_executor_select_action.on_execution_select_actions(executor="Test_str_type")
+
+
+# Test on_execution_select_action_compute
+def test_on_execution_select_action_compute(
+    mock_executor: Executor, dummy_config: type
+) -> None:
+    """Test on_execution_select_action_compute.
+
+    Args:
+        dummy_config: config
+        mock_executor: Executor
+    """
+    ff_executor = FeedforwardExecutorSelectAction(dummy_config)
+    ff_executor.on_execution_select_action_compute(executor=mock_executor)
+    assert mock_executor.store.action_info == "action_info_after_get_action"
+    assert mock_executor.store.policy_info == "policy_info_after_get_action"
```

Comment:
This does not test if the correct network was retrieved from the store (see `action_selection.py` line 95), since the output is the same no matter which network is retrieved

Reply:
I will try to make changes on the mock so that it works based on our goal